sw/device/lib/crypto/impl/status.c

```diff
@@ -0,0 +1,66 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/impl/status.h"
+
+#include "sw/device/lib/base/hardened_status.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/crypto/include/datatypes.h"
+
+crypto_status_t crypto_status_interpret(status_t status) {
+  // First, check for a hardened-ok status.
+  uint32_t res = launder32(kCryptoStatusOK ^ kHardenedBoolTrue);
+  hardened_bool_t is_ok = hardened_status_ok(status);
+  if (launder32(is_ok) == kHardenedBoolTrue) {
+    HARDENED_CHECK_EQ(is_ok, kHardenedBoolTrue);
+    return res ^ is_ok;
+  }
+  HARDENED_CHECK_NE(is_ok, kHardenedBoolTrue);
```

Comment:
nit: this might be simpler:
```
res = launder(kCryptoOk ^ kHardenedTrue);
barrier32(res);
res ^= hardened_status_ok(status);
if (launder32(res) == kHardenedBoolTrue) {
  // we should assert that hardenedTrue == cryptoOk if we don't have an assertion already.
  return (crypto_status_t)res;
}

...

```